tests/integration/generic_test.go

```diff
@@ -106,48 +109,45 @@ var _ = Describe("odo generic", func() {
 		})
 	})
 
-	Context("Delete the project with flag -o json", func() {
+	When("deleting a project with flag -o json", func() {
 		var projectName string
 		JustBeforeEach(func() {
 			projectName = helper.RandString(6)
 		})
-
-		// odo project delete foobar -o json
-		It("should be able to delete project and show output in json format", func() {
+		It("should delete the project and show output in json format", func() {
 			helper.Cmd("odo", "project", "create", projectName, "-o", "json").ShouldPass()
-
 			actual := helper.Cmd("odo", "project", "delete", projectName, "-o", "json").ShouldPass().Out()
 			valuesDel := gjson.GetMany(actual, "kind", "message")
 			expectedDel := []string{"Status", "Deleted project"}
 			Expect(helper.GjsonMatcher(valuesDel, expectedDel)).To(Equal(true))
-
 		})
+
 	})
 
-	Context("When deleting two project one after the other", func() {
-		It("should be able to delete sequentially", func() {
+	When("deleting two project one after the other", func() {
+		It("should be able to delete them sequentially", func() {
 			project1 := helper.CreateRandProject()
 			project2 := helper.CreateRandProject()
 
-			helper.DeleteProject(project2)
 			helper.DeleteProject(project1)
+			helper.DeleteProject(project2)
 		})
 	})
 
-	Context("When deleting three project one after the other in opposite order", func() {
-		It("should be able to delete", func() {
+	When("deleting several projects", func() {
+		It("should be able to delete them in any order", func() {
 			project1 := helper.CreateRandProject()
 			project2 := helper.CreateRandProject()
 			project3 := helper.CreateRandProject()
 
-			helper.DeleteProject(project1)
 			helper.DeleteProject(project2)
+			helper.DeleteProject(project1)
 			helper.DeleteProject(project3)
 
 		})
 	})
 
-	Context("when executing odo version command", func() {
+	When("executing odo version command", func() {
```

Comment:
Same comment as above. Either add a `BeforeEach` block to continue using `When`, or remove it, but ensure you modify your `It` statement accordingly.